.pre-commit-config.yaml

```diff
@@ -0,0 +1,87 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+# Linting
+- repo: https://github.com/pre-commit/pre-commit-hooks
+  rev: v4.3.0
+  hooks:
+  - id: trailing-whitespace
+  - id: end-of-file-fixer
+    exclude_types: [jupyter]
+  - id: check-toml
+  - id: check-added-large-files
+  - id: check-ast
```

Comment:
I would also put check-ast first and (if possible) stop the execution of the hook. If you code is not valid Python what are you even doing.

Reply:
Moved `check-ast` first, good suggestion.

Is there an existing hook for circular imports? I don't see anything in ~1m Googling but happy to add it if it exists, would be useful!

Reply:
There's https://github.com/bndr/pycycle which I've used in the past and we could run as a local hook, but it hasn't been maintained for a relatively long time, so maybe not a great idea to add it to pre-commit. We could add it for local checks but not the CI I guess...

 It worked fine every time, and I don't expect that Python's import system has changed in the last 5 years or will change, but generally hesitant to use stuff that's not actively maintained.

Reply:
Oh, that's a cool project, shame it's not actively maintained. I'd definitely be in favour of adding it if it was maintained, but otherwise might be more risk than it's worth. Cycles should at least trigger an error in tests (when they import us), so at least they won't slip through undetected. Ultimately comes down to how often we end up running into cyclical import errors -- if it's common then feel free to open another PR to add it :)